src/FuncTorrent/ControlThread.hs

```diff
@@ -0,0 +1,209 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module FuncTorrent.ControlThread where
+
+import Control.Concurrent
+import GHC.Conc
+import Control.Lens
+import Data.IORef
+import Control.Monad hiding (
+    forM , forM_ , mapM , mapM_ , msum , sequence , sequence_ )
+
+import FuncTorrent.Tracker (TrackerResponse(..), tracker, mkTrackerResponse, peers)
+import FuncTorrent.Bencode (decode)
+import FuncTorrent.Metainfo (Metainfo(..))
+
+import FuncTorrent.Peer
+import FuncTorrent.PeerThread
+import FuncTorrent.PeerThreadData
+
+data ControlThread = ControlThread {
+        _metaInfo           :: Metainfo
+    ,   _trackerResponses   :: [TrackerResponse]
+    ,   _peerList           :: [Peer]
+    ,   _peerThreads        :: [(PeerThread, ThreadId)]
+--    ,   _diskIO_Handle   :: Handle
+    ,   _controlTStatus     :: IORef ControlThreadStatus
+    ,   _controlTAction     :: IORef ControlThreadAction
+    }
+
+data ControlThreadStatus =
+        Stopped
+    |   Downloading
+    |   Seeding
+  deriving (Eq, Show)
```

Comment:
Any particular reason for `deriving (Eq, Show)` ? Same question about `ControlThreadAction`